### PR TITLE
[ci] fix R linting job (fixes #3873)

### LINKED ISSUE #3873: R-lintr failures at CI
Our R-lintr CI is broken. I see this error in `master`, #3866 and #3872.

https://github.com/microsoft/LightGBM/runs/1783657796

```
Warning message:
package ‘lintr’ was built under R version 3.6.3 
[1] 49 R files need linting
Error in loadNamespace(j <- i[[1L]], c(lib.loc, .libPaths()), versionCheck = vI[[j]]) : 
  namespace ‘xfun’ 0.15 is being loaded, but >= 0.19 is required
Calls: <Anonymous> ... namespaceImportFrom -> asNamespace -> loadNamespace
Execution halted
```

### PR DESCRIPTION
**Pull request by jameslamb**
Attempting to fix #3873.

> Warning message:
package ‘lintr’ was built under R version 3.6.3 
[1] 49 R files need linting
Error in loadNamespace(j <- i[[1L]], c(lib.loc, .libPaths()), versionCheck = vI[[j]]) : 
  namespace ‘xfun’ 0.15 is being loaded, but >= 0.19 is required
Calls: <Anonymous> ... namespaceImportFrom -> asNamespace -> loadNamespace
Execution halted

There might be a mistake somewhere in `lintr`'s feedstock on conda-forge or in one of its dependencies. `{xfun}` is not a direct dependency of `{lintr}` (https://github.com/conda-forge/r-lintr-feedstock/blob/master/recipe/meta.yaml), so it must be a dependency of a dependency (or deeper :laughing: ). For now, I'm just going to try forcing `conda install` to get a newer version.

### CODE REVIEW

**Comment by jameslamb** [comment]
It seems like this fix did fix the lint job! https://github.com/microsoft/LightGBM/pull/3877/checks?check_run_id=1785176162

File: .ci/test.sh [hunk 1]
```diff
@@ -44,9 +44,11 @@ if [[ $TASK == "lint" ]]; then
         pycodestyle \
         pydocstyle \
         r-stringi  # stringi needs to be installed separate from r-lintr to avoid issues like 'unable to load shared object stringi.so'
+    # r-xfun below has to be upgraded because lintr requires > 0.19 for that package
     conda install -q -y -n $CONDA_ENV \
         -c conda-forge \
             libxml2 \
+            "r-xfun>=0.20" \
```
Comment (jameslamb) [inline .ci/test.sh:51]:
```suggestion
            "r-xfun>=0.19" \
```

**Comment by StrikerRUS** [comment]
Should we report this in conda-forge repo?

**Comment by jameslamb** [comment]
> Should we report this in conda-forge repo?

yes, I will

**Comment by jameslamb** [comment]
> should we report this in the conda-forge repo

Created https://github.com/conda-forge/r-lintr-feedstock/issues/12

**Comment by jameslamb** [comment]
It seems that `lint` is passing, and all other CI jobs are passing except the ones we know are broken and waiting on other PRs (https://github.com/microsoft/LightGBM/pull/3876#issuecomment-769163119), so I'm going to merge this.

**Comment by StrikerRUS** [comment]
@jameslamb 
> Created conda-forge/r-lintr-feedstock#12

Thank you very much! Hope this will allow us revert the workaround eventually.

**Pull request merged**